packages/repository/src/repositories/repository.ts

```diff
@@ -26,7 +26,37 @@ export interface Repository<T extends Model> {}
 
 export interface ExecutableRepository<T extends Model> extends Repository<T> {
   /**
-   * Execute a query with the given parameter object or an array of parameters
+   * Proxy method for the underlying connector's method to execute raw queries.
+   * Support and method signature are entirely dependent on the underlying
+   * connector's implementation. The result object is the connector's native
+   * result object, you will have to process it on your own.
+   *
+   * @remarks
+   *
+   * MySQL usage example:
+   *
+   * Although the whole command can be specified in the `command` argument. It
+   * is recommended to specify the parameters in the `parameters` argument,
+   * which is an array in case of MySQL.
+   *
+   *```ts
+   * myRepository.execute(
+   *  'SELECT * from User WHERE name=? && status=?',
+   *  ['John', 'active']
+   *);
+   *```
+   *
+   * Using `parameters` argument helps to prevent SQL injection attacks.
+   *
+   * MongoDB usage example:
+   *
+   * In case of MongoDB, the first parameter is the collection name, the second
+   * parameter is the command name, and the third being the options.
+   *
+   * const res = await myRepository.execute(
+   *   'Pet', 'find', {type:'cat'}
+   * );
```

Comment:
Are you sure the string `find` is assignable to `NamedParameters | PositionalParameters`?

ATM, `ExecutableRepository.execute` provides SQL-like flavor only:

```ts
interface ExecutableRepository { 
  execute(
    command: Command,
    parameters: NamedParameters | PositionalParameters,
    options?: Options,
  ): Promise<AnyObject>;
}
```

To support other connectors, I had to add few more overloads:

```ts
interface ExecutableRepository { 
  // MongoDB
  execute(
     collectionName: string,
     command: string,
     ...parameters: PositionalParameters
   ): Promise<AnyObject>;

  // Other connectors
  execute(...args: PositionalParameters): Promise<AnyObject>;
}
```